From 083920b49245b7c091a97023320222b3a40f0df7 Mon Sep 17 00:00:00 2001
From: Andrei Gherzan <andrei@resin.io>
Date: Mon, 20 May 2019 10:37:41 +0000
Subject: [PATCH 5/5] dont-start-services-in-container

When resinOS is ran as privileged container, avoid messing with tty1 as it will
interfere with the host.

Upstream-status: Innappropiate [Configuration]
Signed-off-by: Andrei Gherzan <andrei@resin.io>
---
 systemd-units/plymouth-quit-wait.service.in         | 1 +
 systemd-units/plymouth-quit.service.in              | 1 +
 systemd-units/plymouth-read-write.service.in        | 1 +
 systemd-units/plymouth-switch-root.service.in       | 1 +
 systemd-units/systemd-ask-password-plymouth.path.in | 1 +
 5 files changed, 5 insertions(+)

diff --git a/systemd-units/plymouth-quit-wait.service.in b/systemd-units/plymouth-quit-wait.service.in
index fbb235c..a98c706 100644
--- a/systemd-units/plymouth-quit-wait.service.in
+++ b/systemd-units/plymouth-quit-wait.service.in
@@ -1,6 +1,7 @@
 [Unit]
 Description=Hold until boot process finishes up
 After=rc-local.service plymouth-start.service systemd-user-sessions.service
+ConditionVirtualization=!container
 
 [Service]
 ExecStart=-@PLYMOUTH_CLIENT_DIR@/plymouth --wait
diff --git a/systemd-units/plymouth-quit.service.in b/systemd-units/plymouth-quit.service.in
index cf9901e..6b72c58 100644
--- a/systemd-units/plymouth-quit.service.in
+++ b/systemd-units/plymouth-quit.service.in
@@ -1,6 +1,7 @@
 [Unit]
 Description=Terminate Plymouth Boot Screen
 After=rc-local.service plymouth-start.service systemd-user-sessions.service
+ConditionVirtualization=!container
 
 [Service]
 ExecStart=-@PLYMOUTH_CLIENT_DIR@/plymouth quit
diff --git a/systemd-units/plymouth-read-write.service.in b/systemd-units/plymouth-read-write.service.in
index 55b975e..2539b0b 100644
--- a/systemd-units/plymouth-read-write.service.in
+++ b/systemd-units/plymouth-read-write.service.in
@@ -4,6 +4,7 @@ DefaultDependencies=no
 After=local-fs.target
 Before=sysinit.target
 ConditionPathExists=!/etc/initrd-release
+ConditionVirtualization=!container
 
 [Service]
 ExecStart=-@PLYMOUTH_CLIENT_DIR@/plymouth update-root-fs --read-write
diff --git a/systemd-units/plymouth-switch-root.service.in b/systemd-units/plymouth-switch-root.service.in
index 8986655..f8bbb92 100644
--- a/systemd-units/plymouth-switch-root.service.in
+++ b/systemd-units/plymouth-switch-root.service.in
@@ -3,6 +3,7 @@ Description=Plymouth switch root service
 DefaultDependencies=no
 ConditionPathExists=/etc/initrd-release
 Before=initrd-switch-root.service
+ConditionVirtualization=!container
 
 [Service]
 Type=oneshot
diff --git a/systemd-units/systemd-ask-password-plymouth.path.in b/systemd-units/systemd-ask-password-plymouth.path.in
index c2ee420..826a6b3 100644
--- a/systemd-units/systemd-ask-password-plymouth.path.in
+++ b/systemd-units/systemd-ask-password-plymouth.path.in
@@ -7,6 +7,7 @@ After=plymouth-start.service
 Before=basic.target shutdown.target
 ConditionKernelCommandLine=!plymouth.enable=0
 ConditionPathExists=/run/plymouth/pid
+ConditionVirtualization=!container
 
 [Path]
 DirectoryNotEmpty=/run/systemd/ask-password
-- 
2.7.4

